### ISSUE TITLE
Add support for REST-AT

### ISSUE THREAD

**Issue by tomsontom** [quarkusio/quarkus#15364]
**Description**
I'm trying to gradually break apart a big EE application who relies on distributed transactions but I also think that there are situations where using 2pc setup is better suited than using compensations or the use of the saga-pattern. So having support for REST-AT built into Quarkus would be a great addition to the toolset of developers.

**Implementation ideas**
I started a thread some time ago at the Narayana user group [1] and @mmusgrov gave some interesting pointers and even hacked together a working solution.

[1] https://groups.google.com/g/narayana-users/c/acJ_7Us6JAI

**Comment by famod** [comment]
Very interesting proposal! I read that mail thread briefly and I must say that I haven't fully understood whether this can/should be a (quarkiverse?) extension or some core extension?

/cc @gsmet @Sanne @geoand

**Comment by geoand** [comment]
I don't really have an opinion on this TBH

**Comment by gsmet** [comment]
Well, before even deciding where it should be, we would need someone to develop and maintain the extension :).

**Comment by Sanne** [comment]
Right. Looks very interesting but it depends on the Transactions team to be able to develop & maintain this. cc/ @ochaloup @mmusgrov and @tomjenkinson

**Comment by mmusgrov** [comment]
We think we may have a resource to work on this task. We will keep you informed when we have a firm commitment and have planned out the work.

**Comment by mmusgrov** [comment]
There is some preparatory work that we need to do in narayana before we can provide this extension which we have captured in a [JIRA issue](https://issues.redhat.com/browse/JBTM-3457). Sweta is doing the work and is making good progress.

**Comment by geoand** [comment]
@mmusgrov any update on this?

**Comment by mmusgrov** [comment]
@geoand we have all the bits that we need so we just need to pull them together on the narayana side and then we can start work on providing a quarkus extension. Is there any particular priority level for doing this work?

**Comment by geoand** [comment]
Thanks for the info.

I personally am not aware of any specific priority for this

**Comment by mmusgrov** [comment]
Ok but it is definitely something that we wish to provide.

**Comment by mmusgrov** [comment]
We use rts, short for RESTful transaction service, to refer to any REST API to our transaction models, so far we have at least two:
- the implementation of the MP-LRA spec implementation has maven coordinates `org.jboss.narayana.rts:narayana-lra org.jboss.narayana.rts:restat`
- the REST API to the JTA transaction manager has maven coordinates `org.jboss.narayana.rts:restat`
